tests/unit/test_server_class/test_manifest.py

```diff
@@ -0,0 +1,45 @@
+import pytest
+
+from renku_notebooks.api.classes.server import UserServer
+
+
+@pytest.mark.parametrize(
+    "parameters,expected",
+    [
+        (
+            {"environment_variables": {"TEST": "testval"}},
+            "/containers/0/env/-', 'value': {'name': 'TEST', 'value': 'testval'}",
+        )
+    ],
+)
+def test_session_manifest(
```

Comment:
This is a really nice way to test a decent portion of the server class functionality. 

It would be nice to add a few lines (or a separate test) to check the opposite works. I.e. that you can call `UserServer.from_js(user, manifest)` and get the environment variables in the `.environment_variables` on the server instance. 

This is how I quickly attempted to modify the test for this case and how I caught the missing `/` from the other comment. But I think you made this function to be more general based on the naming. So we can either rename the function to only test the patching and parsing env variables.


```python
@pytest.mark.parametrize(
    "parameters,expected",
    [
        (
            {"environment_variables": {"TEST": "testval"}},
            "/containers/0/env/-', 'value': {'name': 'TEST', 'value': 'testval'}",
        )
    ],
)
def test_session_manifest(
    parameters, expected, patch_user_server, user_with_project_path, app
):
    """Test that session manifest can be created correctly."""

    user = user_with_project_path("namespace/project")
    with app.app_context():
        base_parameters = {
            "user": user,
            "namespace": "test-namespace",
            "project": "test-project",
            "image": None,
            "server_options": {
                "lfs_auto_fetch": 0,
                "defaultUrl": "/lab",
                "cpu_request": "100",
                "mem_request": "100",
            },
            "branch": "master",
            "commit_sha": "abcdefg123456789",
            "notebook": "",
            "environment_variables": {},
            "cloudstorage": [],
        }

        server = UserServer(**{**base_parameters, **parameters})
        server.image_workdir = ""

        manifest = server._get_session_manifest()
        server_from_manifest = UserServer.from_js(user, manifest)

    assert expected in str(manifest)
    assert len(server_from_manifest.environment_variables) > 0
```